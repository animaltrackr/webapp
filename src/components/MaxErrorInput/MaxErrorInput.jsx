import React, { Component } from 'react';
import Spin from 'antd/lib/spin';
import Icon from 'antd/lib/icon';
import { Slider, InputNumber, Row, Col } from 'antd';

class MaxErrorInput extends Component {
	state = {
		inputValue: this.props.maxError ? this.props.maxError : 0,
	};

	onChange = value => {
		this.props.onMaxErrorRadiusChange(value);
		this.setState({
			inputValue: value,
		});
	};
	render() {
		const loadingIcon = <Icon type="loading" style={{ fontSize: 20 }} spin />;

		const { inputValue } = this.state;
		return (
			<Row>
				<Col span={12}>
					<Slider
						min={0}
						max={100}
						onChange={this.onChange}
						value={inputValue}
					/>
				</Col>
				<Col span={4}>
					<InputNumber
						min={0}
						max={100}
						style={{ marginLeft: 16 }}
						value={inputValue}
						onChange={this.onChange}
					/>
				</Col>
				<Col>
					<Spin
						indicator={loadingIcon}
						hidden={this.props.loading ? !this.props.loading : true}
					/>
				</Col>
			</Row>
		);
	}
}

MaxErrorInput.propTypes = {};

export default MaxErrorInput;
