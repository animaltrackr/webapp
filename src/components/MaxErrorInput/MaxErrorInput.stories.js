import React from 'react';
import { storiesOf } from '@storybook/react';
import MaxErrorInput from './MaxErrorInput';

storiesOf('MaxErrorInput', module).add('MaxErrorInput', () => (
	<div>
		<MaxErrorInput />
	</div>
));
