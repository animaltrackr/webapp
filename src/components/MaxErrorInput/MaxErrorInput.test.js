import React from 'react';
import { shallow } from 'enzyme';
import MaxErrorInput from './MaxErrorInput';

describe('<MaxErrorInput />', () => {
	test('renders', () => {
		const wrapper = shallow(<MaxErrorInput />);
		expect(wrapper).toMatchSnapshot();
	});
});
