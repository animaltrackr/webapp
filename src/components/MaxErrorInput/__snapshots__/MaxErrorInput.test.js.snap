// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaxErrorInput /> renders 1`] = `ShallowWrapper {}`;
